### PR TITLE
Property Test for PoStProof

### PR DESCRIPTION
**Pull request by jdjaustin**
**Summary of changes**
Changes introduced in this pull request:
- Use quickcheck to test our custom implementations of `PoStProof`

**Reference issue to close (if applicable)**

Closes associated task in #1881 

**Other information and links**

N/A

### CODE REVIEW

File: types/src/sector/post.rs
Comment (LesnyRumcajs) [inline types/src/sector/post.rs:157]: Same comment as [here](https://github.com/ChainSafe/forest/pull/1982#discussion_r986615383)

File: types/src/sector/post.rs
Comment (LesnyRumcajs) [inline types/src/sector/post.rs:137]: Could this be entirely inside `mod tests` along with the import?

**Review by LesnyRumcajs — CHANGES_REQUESTED** [review]

**Review by lemmih — APPROVED** [review]

**Review by tyshko5 — APPROVED** [review]

File: types/Cargo.toml
Comment (LesnyRumcajs) [inline types/Cargo.toml:37]: use workspace dependencies?

**Review by LesnyRumcajs — APPROVED** [review]
LGTM, after you merge the changes from the PR introducing workspace dependencies for `quickcheck`, please update the `types/Cargo.toml` and potentially other `.toml` files that would be able to use it.

**Pull request merged**
